backend/functions/lib/adapters/gateways/abstract-mssql-client.ts

```diff
@@ -0,0 +1,112 @@
+import { config, ConnectionError, ConnectionPool, MSSQLError, IResult } from 'mssql';
+import { DbTableFieldSpec, IDbConfig, QueryResults } from '../types/database';
+import { deferClose } from '../../deferrable/defer-close';
+import { ApplicationContext } from '../types/basic';
+import { getCamsError } from '../../common-errors/error-utilities';
+
+export abstract class AbstractMssqlClient {
+  private static connectionPool: ConnectionPool;
+  private readonly moduleName: string;
+
+  protected constructor(context: ApplicationContext, dbConfig: IDbConfig, childModuleName: string) {
+    this.moduleName = `ABSTRACT-MSSQL-CLIENT (${childModuleName})`;
+    if (!AbstractMssqlClient.connectionPool) {
+      AbstractMssqlClient.connectionPool = new ConnectionPool(dbConfig as config);
+      deferClose(AbstractMssqlClient.connectionPool);
+    }
+  }
+
+  public async executeQuery<T = unknown>(
+    context: ApplicationContext,
+    query: string,
+    input?: DbTableFieldSpec[],
+  ): Promise<QueryResults> {
+    // we should do some sanitization here to eliminate sql injection issues
```

Comment:
Is this note even relevant? SQL injection would have had to happen in the query prior to this method being called. Otherwise the input variables are expanded in the query statement by the driver.
```suggestion
```